course2/meeting10/main.cc

```diff
@@ -0,0 +1,54 @@
+#include <random>
+#include <algorithm>
+#include <vector>
+#include <list>
+#include <deque>
+#include <iostream>
+#include <chrono>
+#include <functional>
+
+struct RNG
+{
+	std::default_random_engine randEngine;
+	std::uniform_real_distribution<double> realDist{0, 1000};
+	double operator() () { return realDist(randEngine); }
+};
+
+void PrintTime(std::function<void()> func, const char* message)
+{
+	std::chrono::time_point<std::chrono::system_clock> start, end;
+
+	start = std::chrono::system_clock::now();
+	func();
+	end = std::chrono::system_clock::now();
+
+	std::cout << message << (end-start).count() << std::endl;
+}
+
+
+int main()
+{
+	int N = 1000000;
+	RNG randomGen;
+
+	std::vector<double> v(N);
```

Comment:
Yeah you're right :-) There currently isn't much difference between the three for populating hte containers -- that'll probably be quite different if init is included in the benchmark